File: src/BankMain.java

import java.util.Scanner;

public class BankMain {

    public static void main(String[] args) {
        Bank bank = new Bank("1234");
        Scanner scanner = new Scanner(System.in);
        User loggedInUser = null;

        boolean loggedIn = false;
        while (true) {
            System.out.println("Välkommen till bankappen! Välj ett alternativ:");

            if (loggedIn) {
                System.out.println("1. Skapa konto");
                System.out.println("2. Visa saldo");
                System.out.println("3. Insättning");
                System.out.println("4. Uttag");
                System.out.println("5. Avsluta");
            } else {
                System.out.println("1. Skapa användare");
                System.out.println("5. Avsluta");
                System.out.println("6. Logga in");
            }

            System.out.print("Ange ditt val: ");
            int choice = scanner.nextInt();
            scanner.nextLine(); // Ta bort radbrytningen

            switch (choice) {
                case 1:
                    if (loggedIn) {
                        // Create account
                        System.out.print("Ange användarnamn: ");
                        String username = scanner.nextLine();
                        User user = bank.getUser(username);
                        if (user != null) {
                            System.out.print("Ange kontotyp (t.ex. sparkonto, lönekonto): ");
                            String accountType = scanner.nextLine();
                            System.out.print("Ange räntesats (t.ex. 0.02 för 2%): ");
                            double interestRate = scanner.nextDouble();
                            scanner.nextLine();
                            Account createdAccount = bank.createAccount(accountType, user, interestRate);
                            String accountNumber = createdAccount.getAccountNumber();
                            System.out.println("Konto skapat med kontonummer: " + accountNumber);
                        } else if (user == null) {
                            System.out.println("Användaren hittades inte.");
                        }
                    } else {
                        System.out.print("Ange användarnamn: ");
                        String userId = scanner.nextLine();
                        System.out.print("Ange namn: ");
                        String name = scanner.nextLine();
                        System.out.print("Ange lösenord: ");
                        String password = scanner.nextLine();
                        bank.createUser(userId, name, password);
                        System.out.println("Användare skapad!");
                    }
                    break;

                case 2:
                    // Show balance
                    System.out.print("Ange kontonummer: ");
                    String accountNumber = scanner.nextLine();
                    Account account = bank.getAccount(accountNumber);
                    if (account != null) {
                        System.out.println("Saldo: " + account.getBalance());
                    } else {
                        System.out.println("Kontot hittades inte.");
                    }
                    break;
                case 3:
                    // Deposit
                    System.out.print("Ange kontonummer: ");
                    accountNumber = scanner.nextLine();
                    System.out.print("Ange belopp att sätta in: ");
                    double depositAmount = scanner.nextDouble();
                    scanner.nextLine();
                    account = bank.getAccount(accountNumber);
                    if (account != null) {
                        account.deposit(depositAmount);
                        System.out.println("Insättning genomförd!");
                    } else {
                        System.out.println("Kontot hittades inte.");
                    }
                    break;
                case 4:
                    // Withdrawal
                    System.out.print("Ange kontonummer: ");
                    accountNumber = scanner.nextLine();
                    System.out.print("Ange belopp att ta ut: ");
                    double withdrawalAmount = scanner.nextDouble();
                    scanner.nextLine();
                    account = bank.getAccount(accountNumber);
                    if (account != null) {
                        if (account.withdraw(withdrawalAmount)) {
                            System.out.println("Uttag genomfört!");
                        } else {
                            System.out.println("Otillräckligt saldo för uttag.");
                        }
                    } else {
                        System.out.println("Kontot hittades inte.");
                    }
                    break;
                case 5:
                    System.out.println("Tack för att du använt bankappen!");
                    System.exit(0);
                    break;
                case 6:
                    if (!loggedIn) {
                        // Log in with existing user
                        System.out.print("Ange användarnamn: ");
                        String usernameInput = scanner.nextLine();
                        User userExisting = bank.getUser(usernameInput);
                        if (userExisting != null) {
                            System.out.print("Ange lösenord: ");
                            String passwordExisting = scanner.nextLine();
                            if (userExisting.getPassword().equals(passwordExisting)) {
                                System.out.println("Inloggning lyckades!");
                                loggedInUser = userExisting;
                                loggedIn = true;
                            } else {
                                System.out.println("Fel lösenord. Försök igen.");
                            }
                        } else {
                            System.out.println("Användaren hittades inte.");
                        }
                    }
                    break;
                default:
                    System.out.println("Ogiltigt val. Försök igen.");
            }
        }
    }
}

File: Interface/src/BankMain.js
const readline = require("readline");
const Bank = require("./Bank"); // Assuming you have a separate Bank class implemented

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const bank = new Bank("1234");
let loggedInUser = null;
let loggedIn = false;

function main() {
    console.log("Welcome to the bank app! Choose an option:");

    if (loggedIn) {
        console.log("1. Create account");
        console.log("2. Show balance");
        console.log("3. Deposit");
        console.log("4. Withdrawal");
        console.log("5. Quit");
    } else {
        console.log("1. Create user");
        console.log("5. Quit");
        console.log("6. Log in");
    }

    rl.question("Enter your choice: ", function (choice) {
        switch (parseInt(choice)) {
            case 1:
                if (loggedIn) {
                    // Create account
                    rl.question("Enter username: ", function (username) {
                        const user = bank.getUser(username);
                        if (user) {
                            rl.question("Enter account type (e.g., savings, checking): ", function (accountType) {
                                rl.question("Enter interest rate (e.g., 0.02 for 2%): ", function (interestRate) {
                                    const createdAccount = bank.createAccount(accountType, user, parseFloat(interestRate));
                                    const accountNumber = createdAccount.getAccountNumber();
                                    console.log("Account created with account number: " + accountNumber);
                                    main();
                                });
                            });
                        } else {
                            console.log("User not found.");
                            main();
                        }
                    });
                } else {
                    // Create user
                    rl.question("Enter username: ", function (userId) {
                        rl.question("Enter name: ", function (name) {
                            rl.question("Enter password: ", function (password) {
                                bank.createUser(userId, name, password);
                                console.log("User created!");
                                main();
                            });
                        });
                    });
                }
                break;
            case 2:
                // Show balance
                rl.question("Enter account number: ", function (accountNumber) {
                    const account = bank.getAccount(accountNumber);
                    if (account) {
                        console.log("Balance: " + account.getBalance());
                    } else {
                        console.log("Account not found.");
                    }
                    main();
                });
                break;
            case 3:
                // Deposit
                rl.question("Enter account number: ", function (accountNumber) {
                    rl.question("Enter amount to deposit: ", function (depositAmount) {
                        const account = bank.getAccount(accountNumber);
                        if (account) {
                            account.deposit(parseFloat(depositAmount));
                            console.log("Deposit successful!");
                        } else {
                            console.log("Account not found.");
                        }
                        main();
                    });
                });
                break;
            case 4:
                // Withdrawal
                rl.question("Enter account number: ", function (accountNumber) {
                    rl.question("Enter amount to withdraw: ", function (withdrawalAmount) {
                        const account = bank.getAccount(accountNumber);
                        if (account) {
                            if (account.withdraw(parseFloat(withdrawalAmount))) {
                                console.log("Withdrawal successful!");
                            } else {
                                console.log("Insufficient balance for withdrawal.");
                            }
                        } else {
                            console.log("Account not found.");
                        }
                        main();
                    });
                });
                break;
            case 5:
                console.log("Thank you for using the bank app!");
                rl.close();
                break;
            case 6:
                if (!loggedIn) {
                    // Log in with existing user
                    rl.question("Enter username: ", function (usernameInput) {
                        const userExisting = bank.getUser(usernameInput);
                        if (userExisting) {
                            rl.question("Enter password: ", function (passwordExisting) {
                                if (userExisting.getPassword() === passwordExisting) {
                                    console.log("Login successful!");
                                    loggedInUser = userExisting;
                                    loggedIn = true;
                                } else {
                                    console.log("Incorrect password. Try again.");
                                }
                                main();
                            });
                        } else {
                            console.log("User not found.");
                            main();
                        }
                    });
                }
                break;
            default:
                console.log("Invalid choice. Try again.");
                main();
        }
    });
}

main();